cmd/ping.go

```diff
@@ -0,0 +1,187 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/pkg/console"
+)
+
+const (
+	livelinessEndPoint = "/minio/health/live"
+	protocol           = "http://"
+)
+
+var pingFlags = []cli.Flag{
+	cli.IntFlag{
+		Name:  "count, c",
+		Usage: "will return liveliness for that count number of times and return.",
+	},
+}
+
+// return latency and liveness probe.
+var pingCmd = cli.Command{
+	Name:         "ping",
+	Usage:        "will return latency and liveness probe",
+	Action:       mainPing,
+	Before:       setGlobalsFromContext,
+	OnUsageError: onUsageError,
+	Flags:        append(pingFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET [TARGET...]
+{{if .VisibleFlags}}
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}{{end}}
+EXAMPLES:
+  1. Return Latency and liveness probe.
+     {{.Prompt}} {{.HelpName}} mymino
+
+  2. Return Latency and liveness probe 3 number of times.
+     {{.Prompt}} {{.HelpName}} --count 3 myminio
+`,
+}
+
+// Validate command line arguments.
+func checkPingSyntax(cliCtx *cli.Context) {
+	if !cliCtx.Args().Present() {
+		cli.ShowCommandHelpAndExit(cliCtx, "ping", 1) // last argument is exit code
+	}
+}
+
+type PingResult struct {
+	Pings []Ping `json:"ping,omitempty"`
+}
+
+// Ping message together with fields "Status" and "Error"
+type Ping struct {
+	Server  string `json:"server"`
+	Status  string `json:"status"`
+	Error   string `json:"error,omitempty"`
+	Latency string `json:"latency,omitempty"`
+}
+
+func computeLatency(start time.Time) string {
+	now := time.Now().UTC()
+	diff := now.Sub(start)
+	hours := int(diff.Hours())
+	minutes := int(diff.Minutes()) % 60
+	seconds := int(diff.Seconds()) % 60
+	miliSeconds := int(diff.Milliseconds())
+	microSeconds := int(diff.Microseconds())
+	nanoSeconds := int(diff.Nanoseconds())
+	if hours == 0 {
+		if minutes == 0 {
+			if seconds == 0 {
+				if miliSeconds == 0 {
+					if microSeconds == 0 {
+						return fmt.Sprint(nanoSeconds, " nano seconds")
+					}
+					return fmt.Sprint(microSeconds, " micro seconds")
+				}
+				return fmt.Sprint(miliSeconds, " mili seconds")
+			}
+			return fmt.Sprint(seconds, " seconds")
+		}
+		return fmt.Sprint(minutes, " minutes")
+	}
+	return fmt.Sprint(hours, " hours")
+}
```

Comment:
```suggestion
func computeLatency(start time.Time) string {
	diff := time.Since(start)
	hours := int(diff.Hours())
	minutes := int(diff.Minutes()) % 60
	seconds := int(diff.Seconds()) % 60
	milliSeconds := int(diff.Milliseconds())
	microSeconds := int(diff.Microseconds())
	nanoSeconds := int(diff.Nanoseconds())
	switch {
	case hours > 0:
		return english.Plural(hours, "hour", "")
	case minutes > 0:
		return english.Plural(minutes, " minute", "")
	case seconds > 0:
		return english.Plural(seconds, " second", "")
	case milliSeconds > 0:
		return english.Plural(milliSeconds, " millisecond", "")
	case microSeconds > 0:
		return english.Plural(microSeconds, " microsecond", "")
	default:
		return english.Plural(nanoSeconds, " nanosecond", "")
	}
}
```